payments_payu/provider.py

```diff
@@ -547,29 +552,32 @@ def process_notification(self, payment, request):
                     print(refunded_price, payment.total)
                     if data["refund"]["status"] == "FINALIZED":
                         payment.message += data["refund"]["reasonDescription"]
-                        if refunded_price == payment.total:
+                        if refunded_price == payment.captured_amount:
                             payment.change_status(PaymentStatus.REFUNDED)
                         else:
-                            payment.total -= refunded_price
+                            payment.captured_amount -= refunded_price
                             payment.save()
```

Comment:
I think that this is more aligned with https://github.com/jazzband/django-payments/blob/1222e0ab5ec51b525f5d0033833a67ae702df296/payments/models.py#L222 and https://github.com/jazzband/django-payments/blob/1222e0ab5ec51b525f5d0033833a67ae702df296/payments/paypal/__init__.py#L314

Although, the whole behavior of `Payment.refund` is probably still not clear: https://github.com/jazzband/django-payments/issues/399